### DISCUSSION TITLE
Error: Schema must contain uniquely named types but contains multiple types named "hasura_parties"

### DISCUSSION THREAD

**Discussion by ccalvarez** [gatsbyjs/gatsby#32385] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Trying to customize a Hasura GraphQL Type, following the [Schema Customization  API](https://www.gatsbyjs.com/docs/reference/graphql-data-layer/schema-customization/) guide, throws the error:

Error: Schema must contain uniquely named types but contains multiple types named "hasura_parties".

### Reproduction Link

https://github.com/ccalvarez/hasura-type-custom

### Steps to Reproduce

1. Clone the reproducible test case [repo](https://github.com/ccalvarez/hasura-type-custom)
2. cd to folder `hasura-type-custom`
2. Run `npm install` to install dependencies
3. Run `gatsby develop`

### Expected Result

1. The gatsby site 'hello-world' started

2. Able to query the hasura.parties list, including the new field `idToString`, via GraphiQL with this GraphQL query:

`query MyQuery {
  hasura {
    parties {
      id
      name
      idToString
    }
  }
}
`

### Actual Result

1. The gatsby site doesn't start
2. Gatsby build process throws the error:  

Missing onError handler for invocation 'building-schema', error was 'Error: Schema must contain uniquely named types but contains multiple types named "hasura_parties".'.

3. Build process hangs at step ⠇ building schema

### Environment

```shell
System:
    OS: Windows 10 10.0.19042
    CPU: (8) x64 Intel(R) Core(TM) i7-6700HQ CPU @ 2.60GHz
  Binaries:
    Node: 14.17.1 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.5 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.14.13 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.423.0), Chromium (91.0.864.67)
  npmPackages:
    gatsby: ^3.9.1 => 3.9.1 
    gatsby-source-graphql: ^3.9.0 => 3.9.0 
  npmGlobalPackages:
    gatsby-cli: 3.9.0
```

### Config Flags

I don't use any flag.

**Comment by LekoArts** [comment]
If `hasura_parties` already exists and you only want to add `idToString` you shouldn't use the `@dontInfer` flag and add it like this:

```js
exports.createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions
  const typeDefs = `
    type hasura_parties implements Node {
      idToString: String
    }
  `
  createTypes(typeDefs)
}

exports.createResolvers = ({ createResolvers }) => {
  createResolvers({
    hasura_parties: {
      idToString: {
        type: "String",
        resolve: source => source.id.toString(),
      },
    },
  })
}
```

**Reply by ccalvarez** [reply]
@LekoArts  thanks, I have made the change you suggest, I have removed the `@dontInfer` flag, but I'm still getting the same error:

_Error: Schema must contain uniquely named types but contains multiple types named "hasura_parties"._

I have updated the reproduction repo here:  https://github.com/ccalvarez/hasura-type-custom/commit/c6c591dbf76c676b7aea164a2338c41bb3716b9a

I have the same error, either declaring all the attributes of the type:

```js
exports.createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions
  const typeDefs = `
    type hasura_parties implements Node {
      id: Int!,
      name: String!,
      idToString: String
    }
  `
  createTypes(typeDefs)
}
```

or declaring just the new attribute I want to add, as you suggested:

```js
exports.createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions
  const typeDefs = `
    type hasura_parties implements Node {
      idToString: String
    }
  `
  createTypes(typeDefs)
}
```
